plugins/recover.c

```diff
@@ -10,6 +10,91 @@
 #include <plugins/libplugin.h>
 #include <unistd.h>
 
+static struct plugin *plugin;
+static struct plugin_timer *lost_state_timer;
+/* This tells if we are already in the process of recovery. */
+static bool recovery, already_has_peers;
+static void do_check_lost_peer (void *unused);
+static struct node_id local_id;
+
+static void entering_recovery_mode(void *unused) {
+	return;
+}
+
+static struct command_result *after_listpeerchannels(struct command *cmd,
+					             const char *buf,
+					             const jsmntok_t *params,
+					             void *cb_arg UNUSED)
+{
+	plugin_log(plugin, LOG_DBG, "Listpeerchannels called");
+	struct listpeers_channel **channels = json_to_listpeers_channels(tmpctx, buf, params);
+	for (size_t i=0; i<tal_count(channels); i++) {
+		struct listpeers_channel *chan = channels[i];
+		if (chan->lost_state) {
+			plugin_log(plugin, LOG_DBG, "Detected a channel with lost state, Entering Recovery mode!");
+			recovery = true;
+			break;
+		}
+	}
+
+	// TODO: Remove this--
+	recovery = true;
+
+	if (recovery) {
+		entering_recovery_mode(NULL);
+		return command_still_pending(NULL);
+	}
+
+	lost_state_timer = plugin_timer(plugin, time_from_sec(2), do_check_lost_peer, NULL);
+	return command_still_pending(NULL);
+
+}
+
+static struct command_result *check_lost_peer(void *unused) {
+	struct out_req *req;
+	req = jsonrpc_request_start(plugin, NULL, "listpeerchannels",
+					after_listpeerchannels,
+					&forward_error, NULL);
+
+	return send_outreq(plugin, req);
+}
+
+static void do_check_lost_peer (void *unused)
+{
+
+	/* Set to NULL when already in progress. */
+	lost_state_timer = NULL;
+
+	if (recovery) {
+		return;
+	}
+
+	check_lost_peer(unused);
+}
+
+static const char *init(struct plugin *p,
+			const char *buf UNUSED,
+			const jsmntok_t *config UNUSED)
+{
+	plugin = p;
+	plugin_log(p, LOG_DBG, "Recover Plugin Initialised!");
+	recovery = false;
+	lost_state_timer = plugin_timer(plugin, time_from_sec(2), do_check_lost_peer, NULL);
+	u32 num_peers;
+
+	/* Find number of peers */
+	rpc_scan(p, "getinfo",
+		 take(json_out_obj(NULL, NULL, NULL)),
+		 "{id:%,num_peers:%}",
+		 JSON_SCAN(json_to_node_id, &local_id),
+		 JSON_SCAN(json_to_u32, &num_peers));
+
+	already_has_peers = num_peers > 2 ? 1: 0;
+
```

Comment:
Why 2 here not 1?  Deserved a comment?

Reply:
Um I just thought if we're connecting to 10 node in absence of any peers then 3 should be a good threshold to set this field true. nvm I got your point, I'll change it to 1.